packages/flutter_tools/lib/src/android/application_package.dart

```diff
@@ -381,10 +467,18 @@ class ApkManifestData {
       return null;
     }
 
-    final _Attribute? nameAttribute = launchActivity.firstAttribute('android:name');
+    final _Attribute? nameAttribute = isActivityAlias ? launchActivity.firstAttribute('android:targetActivity') : launchActivity.firstAttribute('android:name');
     // "io.flutter.examples.hello_world.MainActivity" (Raw: "io.flutter.examples.hello_world.MainActivity")
     final String? activityName = nameAttribute?.value?.substring(1, nameAttribute.value?.indexOf('" '));
```

Comment:
@andrewkolos , updated

Reply:
I was referring to all logic that serves to find the launch activity (which appears to be everything between [its declaration](https://github.com/SNNafi/flutter/blob/84e109d2d5f0dc7d37870912bc48464ece5cfee8/packages/flutter_tools/lib/src/android/application_package.dart#L179) and when it's [passed to the constructor](https://github.com/SNNafi/flutter/blob/84e109d2d5f0dc7d37870912bc48464ece5cfee8/packages/flutter_tools/lib/src/android/application_package.dart#L249)).

I asked for this because this logic is about 60 lines long, and it's not immediately obvious when it ends. Consider how `fromAndroidProject` would be easier to read if instead of having to read all of https://github.com/SNNafi/flutter/blob/84e109d2d5f0dc7d37870912bc48464ece5cfee8/packages/flutter_tools/lib/src/android/application_package.dart#L178-L237, there was just
```dart
final String? launchActivity = findLaunchActivity(document);
```

Reply:
Oh. I misunderstood then. Will give me a push soon

Reply:
@andrewkolos , Updated